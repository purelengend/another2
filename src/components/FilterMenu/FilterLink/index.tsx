import { memo } from 'react'
import { Link, Text, Image } from '@chakra-ui/react'

interface FilterLinkProps {
  href: string
  icon: string
  label: string
}

const FilterLink = ({ href, icon, label }: FilterLinkProps) => (
  <Link
    href={href}
    className="filter-item"
    fontFamily="Oswald-Regular"
    opacity="0.5"
    _hover={{ textDecoration: 'unset', opacity: '1' }}
  >
    <Image
      src={icon}
      w="68px"
      alt={`The ${label} icon`}
      margin="0 auto"
      loading="eager"
    />
    <Text as="span" display="flex" justifyContent="center">
      {label}
    </Text>
  </Link>
)

export default memo(FilterLink)
