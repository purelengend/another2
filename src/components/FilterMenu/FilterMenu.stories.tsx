import { Meta, Story } from '@storybook/react'
import FilterMenu from './index'

const meta: Meta = {
  title: 'Practice Two/FilterMenu Component',
  component: FilterMenu,
  decorators: [
    Story => (
      <div
        style={{
          backgroundColor: '#532EE7',
          padding: '20px',
          height: '200px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  tags: ['autodocs'],
}

export default meta

const Template: Story<any> = args => <FilterMenu {...args} />

export const Primary = Template.bind({})
Primary.args = {
  isOpen: true,
  customRef: null,
}
