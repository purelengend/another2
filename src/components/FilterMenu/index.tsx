import React, { LegacyRef, memo } from 'react'
import { Flex, Grid } from '@chakra-ui/react'
import tvIcon from '@assets/icons/TV_Icon.svg'
import phoneIcon from '@assets/icons/Phone_Icon.svg'
import laptopIcon from '@assets/icons/Laptop_Icon.svg'
import refrigeratorIcon from '@assets/icons/Refrigerator_Icon.svg'
import airConditionerIcon from '@assets/icons/Air_Conditioner_Icon.svg'
import FilterLink from '@components/FilterMenu/FilterLink'
import { ChevronDownIcon } from '@chakra-ui/icons'

interface FilterMenuProps {
  isOpen: boolean
  customRef: LegacyRef<HTMLDivElement> | undefined
}

const FilterMenu: React.FC<FilterMenuProps> = ({ isOpen, customRef }) => (
  <Flex
    className="filter-menu"
    display={isOpen ? 'block' : 'none'}
    position="relative"
    ref={customRef}
    maxWidth={{
      sm: '340px',
      md: '720px',
      lg: '940px',
      xl: '1140px',
    }}
    margin="0 auto"
  >
    <ChevronDownIcon
      className="arrow-filter"
      w="27px"
      h="27px"
      top="0"
      left="25px"
      overflow="hidden"
      transform="rotate(45deg)"
      boxShadow="-2px -2px 5px -4px"
      bgColor="white"
      pos="absolute"
    />
    <Grid
      className="filter-list"
      gridTemplateColumns={{ base: 'repeat(2, 1fr)', sm: 'repeat(5, 1fr)' }}
      color="textPrimary"
      bgColor="white"
      zIndex="2"
      pos="absolute"
      top="10px"
      borderRadius="10px"
      justifyContent={{ base: 'space-evenly', sm: 'unset' }}
      textAlign="center"
      gap={{ base: '10px', lg: '20px' }}
      padding={{ base: '5px', md: '15px' }}
      boxShadow="rgba(0, 0, 0, 0.3) 0px 19px 38px, rgba(0, 0, 0, 0.22) 20px 20px 12px"
    >
      <FilterLink href="/tv" icon={tvIcon} label="TV" />
      <FilterLink href="/laptop" icon={laptopIcon} label="Laptop" />
      <FilterLink href="/phone" icon={phoneIcon} label="Phone" />
      <FilterLink
        href="/refrigerator"
        icon={refrigeratorIcon}
        label="Refrigerator"
      />
      <FilterLink
        href="/air_conditioner"
        icon={airConditionerIcon}
        label="Air Conditioner"
      />
    </Grid>
  </Flex>
)

export default memo(FilterMenu)
