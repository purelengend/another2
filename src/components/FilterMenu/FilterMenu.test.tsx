import { render } from '@testing-library/react'
import FilterMenu from '.'

jest.mock('@assets/icons/TV_Icon.svg', () => ({
  default: 'tvIcon',
}))

jest.mock('@assets/icons/Phone_Icon.svg', () => ({
  default: 'phoneIcon',
}))

jest.mock('@assets/icons/Laptop_Icon.svg', () => ({
  default: 'laptopIcon',
}))

jest.mock('@assets/icons/Refrigerator_Icon.svg', () => ({
  default: 'refregeratorIcon',
}))

jest.mock('@assets/icons/Air_Conditioner_Icon.svg', () => ({
  default: 'airConditionerIcon',
}))

describe('FilterMenu Component', () => {
  it('renders with the correct number of FilterLink items', () => {
    const { container } = render(<FilterMenu isOpen={true} customRef={null} />)
    const filterLinks = container.querySelectorAll('.filter-item')
    expect(filterLinks.length).toBe(5)
  })

  it('renders the ChevronDownIcon', () => {
    const { container } = render(<FilterMenu isOpen={true} customRef={null} />)
    const chevronIcon = container.querySelector('.arrow-filter')
    expect(chevronIcon).toBeInTheDocument()
  })

  it('renders FilterLink items with correct labels', () => {
    const { getByText } = render(<FilterMenu isOpen={true} customRef={null} />)
    expect(getByText('TV')).toBeInTheDocument()
    expect(getByText('Laptop')).toBeInTheDocument()
    expect(getByText('Phone')).toBeInTheDocument()
    expect(getByText('Refrigerator')).toBeInTheDocument()
    expect(getByText('Air Conditioner')).toBeInTheDocument()
  })

  it('matches snapshot when open', () => {
    const { container } = render(<FilterMenu isOpen={true} customRef={null} />)
    expect(container).toMatchSnapshot()
  })

  it('matches snapshot when closed', () => {
    const { container } = render(<FilterMenu isOpen={false} customRef={null} />)
    expect(container).toMatchSnapshot()
  })
})
