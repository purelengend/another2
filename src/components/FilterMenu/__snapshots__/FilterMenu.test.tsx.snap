// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu Component matches snapshot when closed 1`] = `
<div>
  <div
    class="filter-menu css-flzwrc"
  >
    <svg
      class="chakra-icon arrow-filter css-cglv4x"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        fill="currentColor"
      />
    </svg>
    <div
      class="filter-list css-7d6vbr"
    >
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/tv"
      >
        <img
          alt="The TV icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          TV
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/laptop"
      >
        <img
          alt="The Laptop icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Laptop
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/phone"
      >
        <img
          alt="The Phone icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Phone
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/refrigerator"
      >
        <img
          alt="The Refrigerator icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Refrigerator
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/air_conditioner"
      >
        <img
          alt="The Air Conditioner icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Air Conditioner
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`FilterMenu Component matches snapshot when open 1`] = `
<div>
  <div
    class="filter-menu css-1lpgmk6"
  >
    <svg
      class="chakra-icon arrow-filter css-cglv4x"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        fill="currentColor"
      />
    </svg>
    <div
      class="filter-list css-7d6vbr"
    >
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/tv"
      >
        <img
          alt="The TV icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          TV
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/laptop"
      >
        <img
          alt="The Laptop icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Laptop
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/phone"
      >
        <img
          alt="The Phone icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Phone
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/refrigerator"
      >
        <img
          alt="The Refrigerator icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Refrigerator
        </span>
      </a>
      <a
        class="chakra-link filter-item css-zkti8d"
        href="/air_conditioner"
      >
        <img
          alt="The Air Conditioner icon"
          class="chakra-image css-1q11q9i"
          loading="eager"
          src="[object Object]"
        />
        <span
          class="chakra-text css-1l4w6pd"
        >
          Air Conditioner
        </span>
      </a>
    </div>
  </div>
</div>
`;
